src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get a list of cars from cars table", e);
+        }
+        cars.forEach(car -> car.setDrivers(getDrivers(car.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String queryUpdateCar = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryUpdateCar)) {
+            statement.setString(1, car.getModel());
+            statement.setObject(2, car.getManufacturer().getId());
+            statement.setObject(3, car.getId());
+            statement.executeUpdate();
+            removeDrivers(car);
+            insertDrivers(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car with id = " + car.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String deleteCarQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(deleteCarQuery)) {
+            statement.setLong(1, carId);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id: " + carId, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, c.model AS car_model, "
+                + "m.id AS manufacturer_id, m.name AS manufacturer_name, "
+                + "m.country AS manufacturer_country "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN drivers d ON d.id = cd.driver_id "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE d.id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            cars.forEach(car -> car.setDrivers(getDrivers(car.getId())));
```

Comment:
here you still have nested connections issue. Please move this `cars.forEach(car -> car.setDrivers(getDrivers(car.getId())));` out of the current try catch block